front/components/agent_builder/settings/TagsSelector.tsx

```diff
@@ -40,154 +39,210 @@ export const TagsSelector = ({
-              <DropdownMenuSearchbar
-                autoFocus
-                placeholder="Choose an option"
-                name="input"
-                value={searchText}
-                onChange={setSearchText}
-                onKeyDown={(e) => {
-                  if (e.key === "Enter" && filteredTags.length > 0) {
-                    const firstUnselected = filteredTags.find(
-                      (tag) => !currentTagIds.has(tag.sId)
-                    );
-                    if (firstUnselected) {
-                      onAddTag(firstUnselected);
+    <div className="space-y-2">
+      <div className="flex gap-2">
+        <DropdownMenu
+          open={isMenuOpen}
+          onOpenChange={onMenuOpenChange}
+          modal={false}
+        >
+          <DropdownMenuTrigger asChild>
+            <Button
+              icon={PlusIcon}
+              variant="outline"
+              label="Add"
+              isSelect
+              size="sm"
+              tooltip="Select a tag"
+            />
+          </DropdownMenuTrigger>
+          <DropdownMenuContent className="w-96" side="top" align="start">
+            <DropdownMenuSearchbar
+              autoFocus
+              placeholder="Choose an option"
+              name="input"
+              value={searchText}
+              onChange={setSearchText}
+              onKeyDown={async (e) => {
+                if (e.key === "Enter") {
+                  if (showCreateOption) {
+                    await handleCreateTag(searchText.trim());
+                  } else {
+                    const allAvailable = [
+                      ...suggestedFilteredTags,
+                      ...otherFilteredTags,
+                    ];
+                    if (allAvailable.length > 0) {
+                      onAddTag(allAvailable[0]);
                     }
                   }
-                }}
-              />
-              <DropdownMenuSeparator />
-              <div className="max-h-80 overflow-auto">
-                {filteredTags.map((tag) => (
-                  <DropdownMenuCheckboxItem
-                    key={tag.sId}
-                    label={tag.name}
-                    checked={currentTagIds.has(tag.sId)}
-                    onCheckedChange={(checked) => {
-                      if (checked) {
-                        onAddTag(tag);
-                      } else {
-                        onRemoveTag(tag.sId);
-                      }
-                    }}
-                    onSelect={(event) => {
-                      event.preventDefault();
-                    }}
-                  />
-                ))}
-              </div>
-              {(isAdmin(owner) || onSuggestTags) && (
+                }
+              }}
+            />
+            <DropdownMenuSeparator />
+            <div className="max-h-80 overflow-auto">
+              {showCreateOption && (
+                <DropdownMenuItem
+                  label={`Create "${searchText.trim()}"`}
```

Comment:
nit: wouldn't it make sense to just do `onChange={(text) => setSearchText(text.trim())}` once not have to think to do it everywhere in the code ?

Reply:
Hmmm this would mean we won't be able to add any whitespace right?